pkg/iac/scanners/helm/test/parser_test.go

```diff
@@ -114,13 +114,13 @@ func Test_tar_is_chart(t *testing.T) {
 
 		t.Logf("Running test: %s", test.testName)
 		testPath := filepath.Join("testdata", test.archiveFile)
-		file, err := os.Open(testPath)
-		defer func() { _ = file.Close() }()
-		require.NoError(t, err)
-
-		assert.Equal(t, test.isHelmChart, detection.IsHelmChartArchive(test.archiveFile, file))
+		func() {
+			file, err := os.Open(testPath)
+			require.NoError(t, err)
+			defer file.Close()
 
-		_ = file.Close()
+			assert.Equal(t, test.isHelmChart, detection.IsHelmChartArchive(test.archiveFile, file))
+		}()
```

Comment:
How about this to make them table driven? We could do it for the rest of the tests in this file since we are here. 

```diff
diff --git a/pkg/iac/scanners/helm/test/parser_test.go b/pkg/iac/scanners/helm/test/parser_test.go
index 989590d42..68cc050e4 100644
--- a/pkg/iac/scanners/helm/test/parser_test.go
+++ b/pkg/iac/scanners/helm/test/parser_test.go
@@ -111,16 +111,16 @@ func Test_tar_is_chart(t *testing.T) {
 	}
 
 	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			t.Logf("Running test: %s", test.testName)
+			testPath := filepath.Join("testdata", test.archiveFile)
 
-		t.Logf("Running test: %s", test.testName)
-		testPath := filepath.Join("testdata", test.archiveFile)
-		func() {
 			file, err := os.Open(testPath)
 			require.NoError(t, err)
 			defer file.Close()
 
 			assert.Equal(t, test.isHelmChart, detection.IsHelmChartArchive(test.archiveFile, file))
-		}()
+		})
 	}
 }
 
```

Reply:
Done https://github.com/aquasecurity/trivy/pull/6621/commits/5155dbd9aca2431df54ba27461075772d8f31eb9